### DISCUSSION TITLE
sessionStorage in worker module

### DISCUSSION THREAD

**Discussion by mmohammadi9812** [denoland/deno#13983] [Q&A] [answered]
Hi

I was trying to use `sessionStorage` in worker module, and vscode didn't give me any error

but when I try to run main module (`deno run`), it typecheks the worker module and exists complaining: 
```typescript
error: Uncaught (in worker "") TS2339 [ERROR]: Property 'sessionStorage' does not exist on type 'WorkerGlobalScope & typeof globalThis'.
  self.sessionStorage.setItem("threadId", e.data.threadId.toString());
       ~~~~~~~~~~~~~~
    at file:///Users/mohammad/....some-path...../worker.ts:46:8
```

worker module options are:

```typescript
  const worker = new Worker(new URL("./lib/worker.ts", import.meta.url).href, {
    type: "module",
    deno: {
      namespace: true,
      permissions: {
        net: "inherit",
        read: true,
        write: true,
      }
    }
  });

```

I tried adding `read: true` and `write: true` to worker permissions, and changing `sessionStorage` to `localStorage` to see if anything changes, but none of these helped
deno version is:
```bash
:; deno --version
deno 1.18.2 (release, aarch64-apple-darwin)
v8 9.8.177.6
typescript 4.5.2
```
any thoughts on this?
thanks

**Comment by crowlKats** [comment, marked as answer]
`sessionStorage` & `localStorage` are not available in workers, as per spec

**Reply by mmohammadi9812** [reply]
Oh, I didn't know, thanks for answering
